apps/test-app/frontend/src/components/tree-widget/stateless-v2/TreeModel.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { PresentationInstanceFilterInfo } from "@itwin/presentation-components";
+import { HierarchyNode } from "@itwin/presentation-hierarchy-builder";
+import { InfoNodeTypes } from "./Types";
+
+/** @internal */
+export interface TreeModelRootNode {
+  id: undefined;
+  nodeData: undefined;
+  hierarchyLimit?: number | "unbounded";
+  instanceFilter?: PresentationInstanceFilterInfo;
+}
+
+/** @internal */
+export interface TreeModelHierarchyNode {
+  id: string;
+  nodeData: HierarchyNode;
+  label: string;
+  children: boolean;
+  isLoading?: boolean;
+  isExpanded?: boolean;
+  isSelected?: boolean;
+  hierarchyLimit?: number | "unbounded";
+  instanceFilter?: PresentationInstanceFilterInfo;
+}
+
+/** @internal */
+export interface TreeModelInfoNode {
+  id: string;
+  parentId?: string;
+  type: InfoNodeTypes;
+  message: string;
+}
+
+/** @internal */
+export type TreeModelNode = TreeModelHierarchyNode | TreeModelInfoNode;
+
+/** @internal */
+export function isTreeModelHierarchyNode(node: TreeModelNode): node is TreeModelHierarchyNode {
+  return "nodeData" in node && node.nodeData !== undefined;
+}
+
+/** @internal */
+export interface TreeModel {
+  parentChildMap: Map<string | undefined, string[]>;
+  idToNode: { [id: string]: TreeModelNode };
```

Comment:
Switched both to use `Map`. Using `idToNode.get(nodeId)` instead of `idToNode[nodeId]` not results in `TreeModelNode | undefined` and showed a couple places where I was not checking if node exists in `idToNode` object.